src/main/software/amazon/event/ruler/ComparableNumber.java

```diff
@@ -39,7 +39,29 @@ static String generate(final double f) {
             throw new IllegalArgumentException("Value must be between " + -Constants.FIVE_BILLION +
                     " and " + Constants.FIVE_BILLION + ", inclusive");
         }
-        return String.format("%014x", (long) (TEN_E_SIX * (Constants.FIVE_BILLION + f))).toUpperCase(Locale.ROOT);
+
+        return toHexStringSkippingFirstByte((long) (TEN_E_SIX * (Constants.FIVE_BILLION + f)));
+    }
+
+    private static final String HEXES = "0123456789ABCDEF";
```

Comment:
not a big deal but would love to have this tied to Constants.HEX_DIGITS to avoid one going out to sync with the other.